extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,19 +1,173 @@
 import {Services} from '#service';
 
+import {user} from '#utils/log';
+
 import {getElementConfig} from 'extensions/amp-story/1.0/request-utils';
 
+import {assertHttpsUrl} from '../../../src/url';
 import {
   Action,
   ShoppingConfigDataDef,
 } from '../../amp-story/1.0/amp-story-store-service';
 
+/** @const {!Object<!Object<string, !Array<function>>>} */
+
+const productImagesValidation = {
+  'url': [validateRequired, validateURLs],
+  'alt': [validateRequired, validateString],
+};
+
+const aggregateRatingValidation = {
+  'ratingValue': [validateRequired, validateNumber],
+  'reviewCount': [validateRequired, validateNumber],
+  'reviewUrl': [validateRequired, validateURLs],
+};
+
+export const productValidationConfig = {
```

Comment:
That won't work since `productImagesValidation` and `aggregateRatingValidation` are both used in `productValidationConfig`. If I swap the order, it will say the variables are not defined yet. If there's a workaround, let me know.